### PR TITLE
Implement Open lineage extractor for RedshiftAsync Operators

### LINKED ISSUE #462: Implement OpenLineage extractor for Redshift async operators
Implement OpenLineage extractor for Redshift async operators

- [x] RedshiftSQLOperatorAsync
- [x] RedshiftDataOperatorAsync

### PR DESCRIPTION
**Pull request by rajaths010494**
Export query_ids as XCOM for the Redshift async operators
and add an OpenLineage custom extractor for the same.
Some additional OpenLineage specific environment variables
need to be set which are included as part of the
docker-compose.yaml file change in the commit.
Also adds unit tests for the custom extractor.

Closes #462

### CODE REVIEW

**Comment by bharanidharan14** [comment]
@pankajkoti @phanikumv @pankajastro I request you guys to review this PR even if it is in draft state, it helps to address the review comments faster

File: astronomer/providers/amazon/aws/extractors/redshift.py [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+    SqlJobFacet,
+)
+from openlineage.common.dataset import Dataset, Source
+from openlineage.common.models import DbColumn, DbTableSchema
+from openlineage.common.sql import DbTableMeta, SqlMeta, parse
+
+SCHEMA_URI = "https://raw.githubusercontent.com/OpenLineage/OpenLineage/main/spec/OpenLineage.json"
+
+
+class RedshiftAsyncExtractor(BaseExtractor):
```
Comment (bharanidharan14) [inline astronomer/providers/amazon/aws/extractors/redshift.py:21]: which RedshiftAsyncExtractor ? is it SQL or Cluster or Redshift Data, can add proper name for this extractor class?

File: astronomer/providers/amazon/aws/extractors/redshift.py [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+    """This facet represents a RedshiftFacets"""
+
+    run_facets: Dict[str, BaseFacet] = attr.ib()
+    inputs: List[Dataset] = attr.ib()
+    output: List[Dataset] = attr.ib()
+
+
+# To be removed once redshift extractors are released in openlineage
+class RedshiftDataDatasetsProvider:
+    """RedshiftDataDatasetsProvider"""
```
Comment (bharanidharan14) [inline astronomer/providers/amazon/aws/extractors/redshift.py:119]: Need proper Doc string

File: astronomer/providers/amazon/aws/extractors/redshift.py [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+
+    @staticmethod
+    def _get_schema() -> str:
+        return SCHEMA_URI + "#/definitions/ErrorMessageRunFacet"
+
+
+# To be removed once redshift extractors are released in openlineage
+@attr.s
+class RedshiftFacets:
+    """This facet represents a RedshiftFacets"""
```
Comment (bharanidharan14) [inline astronomer/providers/amazon/aws/extractors/redshift.py:110]: Need clear Doc string, please

File: astronomer/providers/amazon/aws/extractors/redshift.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        """Returns the list of operators this extractor works on."""
+        return ["RedshiftDataOperatorAsync", "RedshiftSQLOperatorAsync"]
+
+    def _get_xcom_redshift_job_id(self, task_instance: TaskInstance) -> Optional[str]:
+        """Get query ids from XCOM"""
+        redshift_job_id: List[str]
+        redshift_job_id = task_instance.xcom_pull(task_ids=task_instance.task_id, key="return_value")
+        if len(redshift_job_id) > 0:
+            return redshift_job_id[0]
+        return None
```
Comment (pankajkoti) [inline astronomer/providers/amazon/aws/extractors/redshift.py:25]: We can remove this line

**Review by pankajkoti — COMMENTED** [review]
Is something pending on this PR?

File: astronomer/providers/amazon/aws/extractors/redshift.py (hunk 1 above)
Comment (rajaths010494) [reply astronomer/providers/amazon/aws/extractors/redshift.py]: It is for Redshift Data and SQL so gave a generic name.

File: astronomer/providers/amazon/aws/extractors/redshift.py (hunk 2 above)
Comment (rajaths010494) [reply astronomer/providers/amazon/aws/extractors/redshift.py]: Removed this as ``openlineage-airflow>=0.12.0`` got released. We don't need this code anymore in our repo.

File: astronomer/providers/amazon/aws/extractors/redshift.py (hunk 3 above)
Comment (rajaths010494) [reply astronomer/providers/amazon/aws/extractors/redshift.py]: Removed this as ``openlineage-airflow>=0.12.0`` got released. We don't need this code anymore in our repo.

File: astronomer/providers/amazon/aws/extractors/redshift.py (hunk 4 above)
Comment (rajaths010494) [reply astronomer/providers/amazon/aws/extractors/redshift.py:25]: mypy throws an issue if i remove it.

**Comment by bharanidharan14** [comment]
@rajaths010494 Is anything pending in this PR?

**Comment by rajaths010494** [comment]
> @rajaths010494 Is anything pending in this PR?

It is ready for review

**Pull request merged**